service/controller/resource/tcnp/create.go

```diff
@@ -294,17 +315,60 @@ func (r *Resource) newAutoScalingGroup(ctx context.Context, cr infrastructurev1a
 		}
 	}
 
+	var maxBatchSize string
+	var minInstancesInService string
```

Comment:
It would be cool to have some structured logging so that we can see certain decisions being made.

Reply:
done